### ISSUE TITLE
Apply opacity to specific tiles 

### ISSUE THREAD

**Issue by rferreiroa** [openseadragon/openseadragon#2170]
Someone have some idea how to start to develop this or to hace control of opacity in certain tiles.

I want to open an image with a certain opacity in all entire image.
When i am a certain zoom, i want to explore all parts of the image and the opacity of the tiles that i have seen is goind dissapear.

Tiles that i havent seen --> certain opacity
Tiles that i have seen --> null opacity.

Some idea to apply certain opacity to specific tiles or the tiles that are on the screen in this moment?

**Comment by iangilman** [comment]
OSD does a lot of work to make it feel like there aren't any tiles and it's all just a single large image. We don't really have a mechanism for manipulating individual tiles. Also, note that if what you have is a tile pyramid, an area that is composed of four tiles when you're zoomed in will be composed of a single tile when you're zoomed out, and a quarter of a tile when you're zoomed out further. For that reason, there aren't really specific tiles for any region (since each region could be covered by a number of different tiles at different levels). 

Can you tell more about the scenario you have in mind? What is significant about the individual tiles that you want to highlight them? How are you intending to address the different zoom levels?

**Comment by selko** [comment]
Of course for this, i think that it is important to develop it in a fixed zoom.

**Comment by iangilman** [comment]
Sounds like a neat feature! May I ask what sort of content you have where you want to keep track of what you have and haven't seen? 

One way I could imagine doing this, which could even support multiple zoom levels, is to have a 2d array of "tiles" which aren't the literal tiles of the image, but just some arbitrary segmentation of the image. Then, as you zoom and pan, you could set values in that array based on the current `viewer.viewport.getBounds`. To avoid setting them all to true at first when you're zoomed out, you could only set them after you're zoomed in a certain amount.

**Comment by selko** [comment]
Yes of course. Are medical images ( histological images). It would be nice to keep track in each slide. 

To have a 2 array of virtual tiles over the origlnal based on the screen width could sounds nice. I'm going to explore this way.
